src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -29,8 +28,17 @@ public static class ResolverUtility
         {
             var key = new LibraryRangeCacheKey(libraryRange, framework);
 
-            return cache.GetOrAdd(key, (cacheKey) =>
-                FindLibraryEntryAsync(cacheKey.LibraryRange, framework, runtimeIdentifier, context, cancellationToken));
+            if (!cache.TryGetValue(key, out var graphItem))
+            {
+                graphItem = FindLibraryEntryAsync(key.LibraryRange, framework, runtimeIdentifier, context, cancellationToken);
+
+                if (!cache.TryAdd(key, graphItem))
```

Comment:
```Suggestion
                cache.TryAdd(key, graphItem);
```
Currently we try to add then it'll fail only if item was already dictionary in dictionary which is very unlikely, even it's the case  those task still return same result. So could you be able to do one more perf profiling with above suggestion and share the result?